extensions/ql-vscode/test/pure-tests/logging.test.ts

```diff
@@ -94,11 +109,36 @@ describe('OutputChannelLogger tests', () => {
     expect(fs.readFileSync(path.join(testLoggerFolder, 'second'), 'utf8')).to.equal('yyy\n');
   });
 
-  it('should delete an existing folder on init', async () => {
+  it('should not remove an additional log location in a custom directory', async () => {
+    logger.setLogStoragePath(tempFolders.storagePath.name, true);
+    await logger.log('xxx', { additionalLogLocation: 'first' });
+    await logger.log('yyy', { additionalLogLocation: 'second' });
+
+    const testLoggerFolder = path.join(tempFolders.storagePath.name, 'test-logger');
+    expect(fs.readdirSync(testLoggerFolder).length).to.equal(2);
+
+    await logger.removeAdditionalLogLocation('first');
+    // need to wait for disposable-object to dispose
+    await waitABit();
+    expect(fs.readdirSync(testLoggerFolder).length).to.equal(2);
+    expect(fs.readFileSync(path.join(testLoggerFolder, 'second'), 'utf8')).to.equal('yyy\n');
+  });
+
+  it('should delete an existing folder when setting the log storage path', async () => {
     fs.createFileSync(path.join(tempFolders.storagePath.name, 'test-logger', 'xxx'));
-    logger.init(tempFolders.storagePath.name);
+    logger.setLogStoragePath(tempFolders.storagePath.name, false);
     // should be empty dir
 
+    const testLoggerFolder = path.join(tempFolders.storagePath.name, 'test-logger');
+    // TODO: Why does this test pass? I'd expect the length to be 0 if it's correctly deleted the existing folder.
+    expect(fs.readdirSync(testLoggerFolder).length).to.equal(1);
+  });
```

Comment:
I couldn't work out why this test passes with "length equal to 1". Can someone explain? I might be misunderstanding what it's doing... 👀

Reply:
Is it possible `fs.remove` is not working because the directory is not empty? Try setting a breakpoint on line 129 and step through to see what happens and what is deleted.

Reply:
Also, I wonder if that method isn't completely synchronous. You can try addinga a `waitABit` call in between and see what happens.

Reply:
Oh the sync/async might be it. I was also baffled by why the existing test worked.